demo-project/src/main/java/com/example/demoproject/DemoController.java

```diff
@@ -26,14 +29,46 @@ public DemoController(Configuration cfg, Function<Configuration,DemoRepository>
      * invokes [f] to execute some business logic with it.
      * After [f] returns successfully, the transaction is co mmitted before returning the result.
      */
-    private <T> T inRepoTransaction(Function<DemoRepository,T> f) {
+    private <T> T inRepoTransaction(Function<DemoRepository, T> f) {
         return ctx.transactionResult(c ->
-            repository.andThen(f).apply(c)
+                repository.andThen(f).apply(c)
         );
     }
 
-    @GetMapping("/values")
+    @GetMapping(value = "/values", produces = MediaType.APPLICATION_JSON_VALUE)
     List<FlywayTest> all() {
         return inRepoTransaction(r -> r.list()).toJavaList();
     }
+
+    @GetMapping(value = "/values/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    ResponseEntity<FlywayTest> findById(@PathVariable("id") Integer id) {
+        FlywayTest ft = inRepoTransaction(r -> r.findById(id));
+        if (ft == null) {
```

Comment:
We haven't talked about `null` yet, but I was intending this to be a "null-free" programming style. In other words, if `r.findById` sometimes doesn't return a value, its return type should be `Optional<T>`, and its documentation should describe when it doesn't return a value.

Reply:
Actually vavr's [Option](https://docs.vavr.io/#_option) would be even more appropriate.